tools/pd-simulator/simulator/task.go

```diff
@@ -401,23 +425,38 @@ func (a *addLearner) IsFinished() bool {
 	return a.finished
 }
 
-func processSnapshot(n *Node, stat *snapshotStat, snapshotSize int64) bool {
-	// If the statement is true, it will start to send or receive the snapshot.
-	if stat.remainSize == snapshotSize {
-		if stat.kind == "sending" {
+func processSnapshot(n *Node, stat *snapshotStat) bool {
+	if stat.status == finished {
+		return true
+	}
+	if stat.status == pending {
+		if stat.action == generate && n.stats.SendingSnapCount > maxSnapGeneratorPoolSize {
+			return false
+		}
+		if stat.action == receive && n.stats.ReceivingSnapCount > maxSnapReceivePoolSize {
+			return false
+		}
+		stat.status = running
+		// If the statement is true, it will start to send or receive the snapshot.
+		if stat.action == generate {
 			n.stats.SendingSnapCount++
 		} else {
 			n.stats.ReceivingSnapCount++
 		}
 	}
-	stat.remainSize -= n.ioRate
-	// The sending or receiving process has not finished yet.
+
+	// store should generate/receive snapshot by chunk size.
+	for n.limiter.AllowN(int(chunkSize)) {
```

Comment:
It cannot be acceleration. we can configure the tick to sped up the simulator.

Reply:
Fix it